server/common/utils/userUtils.js

```diff
@@ -125,11 +128,154 @@ const responseWithUserData = user => {
   return data;
 };
 
+const checkIfUserIsTheOnlyOwner = async (listModel, cohortModel, userId) => {
+  const errors = {};
+
+  const lists = await listModel
+    .find(
+      {
+        $and: [{ ownerIds: userId }, { ownerIds: { $size: 1 } }],
+        'viewersIds.1': { $exists: true }
+      },
+      'name'
+    )
+    .lean()
+    .exec();
+
+  const cohorts = await cohortModel
+    .find(
+      {
+        $and: [{ ownerIds: userId }, { ownerIds: { $size: 1 } }],
+        'memberIds.1': { $exists: true }
+      },
+      'name'
+    )
+    .lean()
+    .exec();
+
+  if (lists.length > 0) {
+    errors.lists = lists;
+  }
+
+  if (cohorts.length > 0) {
+    errors.cohorts = cohorts;
+  }
+
+  return errors;
+};
+
+const deleteAccountDetails = async (userModel, user) => {
+  const { _id: userId, createdAt, email } = user;
+  const hashedEmail = bcrypt.hashSync(email, 12);
+  const removedUser = new User({
+    _id: userId,
+    createdAt,
+    email: hashedEmail,
+    updatedAt: new Date()
+  });
+
+  await userModel.deleteOne({ email });
```

Comment:
This should be a transaction.

Reply:
We know that, but to enable transactions in Mongo we need replicaSet first

Reply:
That should be still possible as @kborucinski had implemented transactions already.